portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +37,49 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Ade!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String comment_text = (String) entity.getProperty("comment_text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment_id = new Comment(id, comment_text, timestamp);
```

Comment:
You can use the local variable name "comment" because it is case-sensitive. So "comment" wouldn't mess up with "Comment" and "comments". So "Comment" means class, "comment" means variable, and "comments" means a list of Comment objects. So the code could look like below:

Comment comment = new Comment(id, commentText, timestamp);
comments.add(comment);